src/server/config/config.js

```diff
@@ -163,8 +187,23 @@ module.exports = class Config {
 
   getSchema() {
     if (!this[schema]) {
-      let objKeys = zipObject([...this[schemaKeys]]);
-      this[schema] = Joi.object().keys(objKeys).default();
+      this[schema] = (function convertToSchema(children) {
+        let objKeys = zipObject([...children]);
```

Comment:
I don't think this is doing what you think it is. Key validation is added per key [here](https://github.com/elastic/kibana/pull/6554/commits/c72c440a1ccf80f4da7f2b0550b1d564809b65fc#diff-00313017c0120ff426b859f2dd5112cdR201).

Since `children` is an object I'm pretty sure that `[...children]` produces `[]`, which means that `zipObject()` produces `{}` and there really isn't any reason for this to happen at all.

If anything we should probably remove the call to `.keys()` on the next line, since it's not doing anything.